src/job-exporter/src/collector.py

```diff
@@ -768,15 +773,31 @@ def __init__(self, name, sleep_time, atomic_ref, iteration_counter):
         Collector.__init__(self, name, sleep_time, atomic_ref, iteration_counter)
 
     def collect_impl(self):
-        process = ps.get_zombie_process(ProcessCollector.cmd_histogram,
+        process_info = ps.get_process_info(ProcessCollector.cmd_histogram,
                 ProcessCollector.cmd_timeout)
 
-        if len(process) > 0:
+        if len(process_info) > 0:
             zombie_metrics = gen_zombie_process_counter()
+            process_mem_metrics = gen_process_mem_usage_gauge()
+            zombie_count = collections.defaultdict(lambda : 0)
+
+            for info in process_info:
+                if info.state == "D":
+                    if "nvidia-smi" in info.cmd:
+                        # override command name to make alert rule easier
+                        zombie_count["nvidia-smi"] += 1
+                    else:
+                        cmd = info.cmd.split()[0] # remove args
+                        zombie_count[cmd] += 1
+
+                if info.rss > 500 * 1024 * 1024:
+                    # only record large memory consumption to save space in prometheus
+                    cmd = info.cmd.split()[0] # remove args
```

Comment:
Here we strip the command details, so most commands will be `bash` or `python`, count them seems meaningless.
How about excluding such processes

Reply:
I removed args because omiagent will also have args like `/opt/omi/bin/omiagent 9 10 --destdir / --providerdir /opt/omi/lib --loglevel WARNING`, I think the args is useless.

Reply:
Why not filter omiagent here, instead of in alert rules.

Reply:
that will be a special case in code